### PR TITLE
✨Electron Skeleton

### PR DESCRIPTION
**Pull request by heikomat**
## What did you change?

~~**⏰ waits on https://github.com/essential-projects/datasource_adapter_localstorage/pull/4**~~

I added a variant of the skeleton that uses the localstorage-datasource-adapter instead of the postgres-one. This variant works without a dedicated database, so it can be used in a standalone electron application

## How can others test the changes?

- enter the `process-engine-electron`-folder
- run `npm install`
- run `npm run build`
- link https://github.com/essential-projects/datasource_adapter_localstorage/pull/4 into that folder
- run `npm start`
- see how you now have a process-engine instance without a dedicated database

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [ ] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: process-engine-electron/bin/start.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  container.validateDependencies();
+
+  try {
+    const bootstrapper = await container.resolveAsync('AppBootstrapper');
+
+    //import default data
+    const datastore = await container.resolveAsync('DatastoreService');
+    const iam = await container.resolveAsync('IamService');
+    const executionContext = await iam.createInternalContext('processengine_system');
```
[the remark this answers is not in the document]
Comment (heikomat) [reply process-engine-electron/bin/start.js]: I didn't define how the method is called, and I don't know if we decided to add the Async-suffix to promise-returning-methods.

That is not a discussion for this or.

File: process-engine-electron/config/development/services/mail.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "from": "processengine@quantusflow.com",
+  "to": "processengine@quantusflow.com"
+}
```
[the remark this answers is not in the document]
Comment (heikomat) [reply process-engine-electron/config/development/services/mail.json]: If we want to send mails without setting a sender, then yes. These are fallback-values

File: process-engine-electron/bin/start.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  for (const iocModule of iocModules) {
+    iocModule.registerInContainer(container);
+  }
+
+  container.validateDependencies();
+
+  try {
+    const bootstrapper = await container.resolveAsync('AppBootstrapper');
+
+    //import default data
```
[the remark this answers is not in the document]
Comment (heikomat) [reply process-engine-electron/bin/start.js]: It is not. The  sole purpose of the following 4 lines is to import the default data. If I didn't want to import the default data, none of the following four lines would be necessary

File: process-engine-electron/config/development/services/mail.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmg-dev) [reply process-engine-electron/config/development/services/mail.json]: do we want to keep those default values?

**Comment by Paulomart** [comment]
@cmg-dev @heikomat Because the blocking PR was merge we can continue here.

**Comment by Paulomart** [comment]
@heikomat I think the changes requested for the README.md by @cmg-dev are not implemented jet.

**Comment by heikomat** [comment]
@cmg-dev Here is an issue for the formatting of the other readmes: #44

**Comment by heikomat** [comment]
@cmg-dev i addressed your comments. Please re-review

**Pull request merged**